### PR TITLE
Update to vyper 0.4

### LINKED ISSUE #1: Add support for Vyper 0.4
Boa currently has the `vyper-0.4.0` branch for vyper 0.4 support.
We should have a similar branch using the [era compiler 1.5.1](https://github.com/matter-labs/era-compiler-vyper/releases/tag/1.5.1)

### PR DESCRIPTION
**Pull request by DanielSchiavini**
Reverts #5, thus reapplying #4. Closes #1

### CODE REVIEW

File: boa_zksync/compiler_utils.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def __boa_debug__() {return_sig}:
 
 def detect_expr_type(source_code, contract):
     ast = parse_to_ast(source_code)
-    vy_ast.folding.fold(ast)
+    # TODO: Do we need to fold? vy_ast.folding.fold(ast)
```
Comment (DanielSchiavini) [inline boa_zksync/compiler_utils.py]: @charles-cooper do we still need this?

**Comment by DanielSchiavini** [comment]
@PatrickAlphaC I contacted the authors of the era test node to figure out why the last tests are failing. However, this version might be already useful for you.

File: boa_zksync/compiler_utils.py (hunk 1 above)
Comment (charles-cooper) [reply boa_zksync/compiler_utils.py]: probably not

Comment (DanielSchiavini) [reply boa_zksync/compiler_utils.py]: Thanks

**Comment by DanielSchiavini** [comment]
I reran everything and it works again. Some rpc instability?

**Pull request merged**